src/main/java/com/gooddata/GoodData.java

```diff
@@ -167,14 +178,30 @@ public GoodData(String hostname, String login, String password, int port, GoodDa
      * @param settings additional settings
      */
     protected GoodData(String hostname, String login, String password, int port, String protocol, GoodDataSettings settings) {
+        this(hostname, login, password, port, protocol, settings, null);
+    }
+
+    /**
+     * Create instance configured to communicate with GoodData Platform running on given host, port and protocol using
+     * given user's credentials.
+     *
+     * @param hostname GoodData Platform's host name (e.g. secure.gooddata.com)
+     * @param login    GoodData user's login
+     * @param password GoodData user's password
+     * @param port     GoodData Platform's API port (e.g. 443)
+     * @param protocol GoodData Platform's API protocol (e.g. https)
+     * @param settings additional settings
+     * @param xGdcHost (optional) X-GDC-HOST header (e.g. giraffes.intgdc.com), when null header is not set
+     */
+    protected GoodData(String hostname, String login, String password, int port, String protocol, GoodDataSettings settings, String xGdcHost) {
```

Comment:
consider some better naming of the new argument for user out of GD who doesn't know our headers (something like `customDomainHostHeader`) and also try to explain in javadoc what is it good for (like that it's being used mainly on testing environments and one can ignore it for production use ;) )